Plugins/SpatialGDK/SpatialGDKEditorToolbar/Source/SpatialGDKEditorToolbar/Private/InteropCodeGenerator/SpatialGDKEditorInteropCodeGenerator.cpp

```diff
@@ -188,63 +211,30 @@ void GenerateInteropFromClasses(const ClassHeaderMap& Classes, const FString& Co
 		const TArray<FString>& TypeBindingHeaders = ClassHeaderList.Value;
 		bool bIsSingleton = SingletonList.Find(ClassHeaderList.Key) != INDEX_NONE;
 
-		ComponentId += GenerateCompleteSchemaFromClass(CombinedSchemaPath, CombinedForwardingCodePath, ComponentId, Class, TypeBindingHeaders, bIsSingleton);
+		ComponentId += GenerateCompleteSchemaFromClass(CombinedSchemaPath, CombinedForwardingCodePath, ComponentId, Class, TypeBindingHeaders, bIsSingleton, Classes);
 	}
 }
 
-bool RunProcess(const FString& Command, const FString& Arguments)
+bool RunProcess(const FString ExecutablePath, FString Arguments)
 {
-	int32 ReturnCode = 0;
-	FString StandardOutput;
-
-	void* ReadPipe = nullptr;
-	void* WritePipe = nullptr;
-	FPlatformProcess::CreatePipe(ReadPipe, WritePipe);
-	FProcHandle ProcHandle = FPlatformProcess::CreateProc(*Command, *Arguments, false, true, true, nullptr, 0, nullptr, WritePipe, ReadPipe);
-
-	if (ProcHandle.IsValid())
+	TSharedPtr<FMonitoredProcess> Process = MakeShareable(new FMonitoredProcess(ExecutablePath, Arguments, true));
+	Process->OnOutput().BindStatic(&OnStatusOutput);
+	Process->Launch();
+	while (Process->Update())
 	{
-		FPlatformProcess::WaitForProc(ProcHandle);
-		StandardOutput = FPlatformProcess::ReadPipe(ReadPipe);
-		FPlatformProcess::GetProcReturnCode(ProcHandle, &ReturnCode);
+		FPlatformProcess::Sleep(0.01f);
```

Comment:
hehe this is a bit gross, could we not use a condition variable or something?

Reply:
I think this is fine.

Reply:
After offline discussion, we will add a comment as the interface appears to be used this way in other places in unreal. The alternative would be to wait for the two OnComplete callbacks before launching the unreal SDK codegen. We stayed with this approach for unreal consistency. However, if we move away SDK codegen, then we should remove the sequential setup we have now and use the callbacks.